src/app/event.service.ts

```diff
@@ -15,35 +16,89 @@ export class EventService implements OnDestroy {
-          setTimeout(() => {
-            this.printing = false;
-          }, 1000);
-        }
-
-        if (event === PrinterEvent.CLOSED) {
-          this.router.navigate(['/standby']);
-        } else if (event === PrinterEvent.CONNECTED) {
-          setTimeout(() => {
-            if (this.configService.isTouchscreen()) {
-              this.router.navigate(['/main-screen']);
-            } else {
-              this.router.navigate(['/main-screen-no-touch']);
-            }
-          }, 1000);
+          this.handlePrinterEvent(event);
         }
       }),
     );
   }
 
+  // https://stackoverflow.com/questions/14425568/interface-type-check-with-typescript
+  // https://stackoverflow.com/questions/8511281/check-if-a-value-is-an-object-in-javascript/8511350#8511350
+  private isPrinterNotification(object: any): object is PrinterNotification {
+    return typeof object === 'object'
+      && object !== null
+      && ('text' in object || 'message' in object || 'action' in object);
+  }
+
+  private handlePrinterNotification(event: PrinterNotification): void {
+    const messages = {
+      'FilamentRunout T0': $localize`:@@prompt-filament-runout-t0:Filament runout detected. Ejecting filament, please wait...`,
+      'Nozzle Parked': $localize`:@@prompt-filament-runout:A filament runout has been detected. Please remove the ejected filament, insert filament from a new spool and press Continue.`,
+      'Continue': $localize`:@@prompt-continue:Continue`,
+      'Paused': $localize`:@@prompt-filament-runout-resume:The filament has been primed. Do you want to continue printing?`,
+      'PurgeMore': $localize`:@@prompt-filament-runout-purge:Purge more filament`,
+      'Heater Timeout': $localize`:@@prompt-heater-timeout:The hotend has been disabled due to inactivity, to avoid burning the filament. Press Reheat when ready to resume.`,
+      'Reheat': $localize`:@@prompt-reheat:Reheat`,
+      'Reheat Done': $localize`:@@prompt-reheat-done:The hotend is now ready.`,
+    };
+    if (event.action === 'close') {
+      this.notificationService.closeNotification();
+    } else if (event.choices?.length > 0) {
+      // event is action:prompt
+      this.notificationService.setPrompt(
+        $localize`:@@action-required:Action required`,
+        messages[event.text] || event.text,
+        event.choices.map(c => messages[c] || c)
+      );
+    } else if (event.choices?.length == 0) {
+      // event is action:prompt without choices
+      this.notificationService.setWarning(
+        $localize`:@@printer-information:Printer information`,
+        messages[event.text] || event.text
+      );
+    } else {
+      // event is action:notification
+      // this.notificationService.setInfo(
+      //   $localize`:@@printer-information:Printer information`,
+      //   messages[event.message] || event.message
+      // );
+      // TODO: annoying as a notification
+      // should be put with an autoclear timeout in the bottom-right statusline
```

Comment:
I think it still should be a notification, just to keep everything consistent. If this isn't too important it might be a good idea to introduce a fourth notification type with a white strip and just set a timeout here to clear the notification after 5s (is that enough time?).

What do you think of that?

Reply:
I need to retest what exactly was being sent, but AFAIR it was every heat enable/disable message (not always coherent with the user actions) and all changes of % being broadcasted, really cluttering information. We have been using it for a month without these and it seems on-point.